backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/MetadataUtils.java

```diff
@@ -677,11 +679,37 @@ public static User loadUserMetadata(SqlDatabase db, String userName) {
     if (userRecord != null) {
       User result = new User(db, userName);
       result.setSettings(userRecord.get(SETTINGS, Map.class));
+      Map<String, String> roles = userRecord.get("roles", Map.class);
+      if (roles != null) {
+        result.setRoles(roles);
+      }
       return result;
     }
     return null;
   }
 
+  public static List<Member> loadUserRoles(SqlDatabase db) {
+    List<Member> members = new ArrayList<>();
+    String roleFilter = Constants.MG_ROLE_PREFIX;
+    String userFilter = Constants.MG_USER_PREFIX;
+    List<Record> result =
+        db.getJooq()
+            .fetch(
+                "select distinct m.rolname as member, r.rolname as role"
+                    + " from pg_catalog.pg_auth_members am "
+                    + " join pg_catalog.pg_roles m on (m.oid = am.member)"
+                    + "join pg_catalog.pg_roles r on (r.oid = am.roleid)"
+                    + "where r.rolname LIKE {0} and m.rolname LIKE {1}",
+                roleFilter + "%", userFilter + "%");
+
+    for (Record r : result) {
+      String memberName = r.getValue("member", String.class).substring(userFilter.length());
+      String roleName = r.getValue("role", String.class).substring(roleFilter.length());
+      members.add(new Member(memberName, roleName));
```

Comment:
Thinking ... I suppose in the graphql you want {_users{name,roles{name}}} or maybe more meaningfull {_users{name,roles{schema,rolename}}}

I didn't check yet what you actually did :-)

Reply:
Didn't do that yet, but already planned to change it do that :)

Reply:
done :)